janus_core/calculations/md.py

```diff
@@ -692,9 +692,13 @@ def __init__(
         super().__init__(ensemble=ensemble, file_prefix=file_prefix, *args, **kwargs)
 
         self.ttime = thermostat_time * units.fs
-        scaled_bulk_modulus = bulk_modulus * units.GPa
+
         if barostat_time:
-            pfactor = (barostat_time * units.fs) ** 2 * scaled_bulk_modulus
+            pfactor = (barostat_time * units.fs) ** 2 * bulk_modulus * units.GPa
+            if self.logger:
+                self.logger.info(
+                    "NPT pfactor=%s GPa fs^2", barostat_time**2 * bulk_modulus
```

Comment:
Why not use `pfactor`? If it's avoiding the unit conversions, couldn't that be hiding any problems with the `pfactor` that's actually used?

Reply:
done